src/index.js

```diff
@@ -138,20 +138,20 @@ class Provider extends React.PureComponent<Props, State> {
     return locale.split('-')[0];
   }
 
-  getAllLocaleData(): Array<Object> {
-    const languages = Object.keys(supportedLocales.reduce((languages, locale) => ({
-      ...languages,
-      [this.localeToLanguage(locale)]: true,
-    }), {}));
+  async getLocaleData(locale: string): Array<Object> {
+    const language = this.localeToLanguage(locale);
 
-    const localeData = languages.map(language => require(`react-intl/locale-data/${language}`));
+    const module = await import(`react-intl/locale-data/${language}`);
+    const localeData = module.default;
 
-    localeData.push({
-      locale: 'tlh',
-      parentLocale: 'en'
-    });
+    if (locale === 'tlh-KL') {
+      localeData.push({
+        locale: 'tlh',
+        parentLocale: 'en',
+      });
+    }
 
-    return localeData.reduce((allLocaleData, localeData) => allLocaleData.concat(localeData), []);
+    return localeData;
```

Comment:
Any flow-typed master in tha house that can solve this? Flow keeps bitching and saying that `localeData` is a Promise instead of an array... Seems like it can't handle `await`.

Reply:
It seems like it's a known issue. https://github.com/facebook/flow/issues/5294
Type it as any for now? 🙈

Reply:
Apparently async functions always return a promise 🤷‍♂️ Fixed it

Reply:
that kinda makes sense when you think about it. kinda.

Reply:
👏